go/protocols/materialize/lifecycle.go

```diff
@@ -158,10 +158,24 @@ type LoadIterator struct {
 	request     *Request        // Request read into.
 	awaitDoneCh <-chan struct{} // Signaled when last commit acknowledgment has completed.
 	err         error           // Terminal error.
+	cancel      context.CancelFunc
+	ctx         context.Context
 }
 
 // Context returns the Context of this LoadIterator.
-func (it *LoadIterator) Context() context.Context { return it.stream.Context() }
+func (it *LoadIterator) Context() context.Context { 
+	if it.ctx == nil {
+		it.ctx, it.cancel = context.WithCancel(it.stream.Context())
```

Comment:
nit: It feels a bit "tricky" to me to lazily construct the context. IMO it would be better to initialize it when the `LoadIterator` is constructed in `transactor.go`. Technically, someone could call `Cancel()` before calling `Context()`, and it would be preferable for `Context()` to return an already-cancelled context in that case.

Reply:
@psFried refactored it